File: engine/engine-model/src/main/scala/tools/packet/guild/bbs/ShowThread.scala
package tools.packet.guild.bbs

import client.database.data.BbsThreadData
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowThread(private var _localThreadId: Int, private var _threadData: BbsThreadData) extends PacketInput {
  def localThreadId: Int = _localThreadId

  def threadData: BbsThreadData = _threadData

  override def opcode(): SendOpcode = SendOpcode.GUILD_BBS_PACKET
}

File: engine/engine-model/src/main/java/tools/packet/guild/bbs/ShowThread.java
package tools.packet.guild.bbs;

import client.database.data.BbsThreadData;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowThread(Integer localThreadId, BbsThreadData threadData) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_BBS_PACKET;
   }
}